### PR TITLE
update pull request template

### PR DESCRIPTION
**Pull request by svrnm**
Updates the pull request template to ask the contributor for sharing some details on their PR. It also let's the contributor know about CI checks, how they can address potential issues and that it is finally not their responsibility to have them fixed perfectly.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!--
+Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to related issues, other PRs, or technical references.
```
Comment (tiffany76) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
Please provide a meaningful description of what your changes will do. Bonus points for including links to related issues, other PRs, or technical references.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+By including a description, help help reviewers to understand the context of your change, which will reduce
+the time to review.
```
Comment (tiffany76) [inline .github/PULL_REQUEST_TEMPLATE.md:5]:
```suggestion
A detailed description helps reviewers understand the context of your change and reduces
the time needed to review.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+Please make sure to follow the [contribution guidelines](https://opentelemetry.io/docs/contributing/).
+
+We will run some checks on your PR. If they fail, take a look in to our [guide on pr-checks](https://opentelemetry.io/docs/contributing/pr-checks/).
+Most issues can be resolved by running `npm run fix:all` locally and pushing the changes.
+But please note that it is not your responsibility to fix all the issues. We will help you with that.
```
Comment (tiffany76) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
We run checks on all PRs. If a check fails on your PR, look for the fix in our [guide](https://opentelemetry.io/docs/contributing/pr-checks/).
Most issues can be resolved by running `npm run fix:all` locally and pushing the changes.
But please note that it is not your responsibility to fix these issues. We will help you.
```

**Review by tiffany76 — APPROVED** [review]
Rewrote a couple parts, but LGTM.

**Review by chalin — COMMENTED** [review]
Whatever suggestions you choose to adopt in https://github.com/open-telemetry/opentelemetry.io/pull/5544#pullrequestreview-2423509985 should be incorporated here too.

**Review by cartermp — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
Will take this one in as-is, it looks good and is an improvement over the other.

**Pull request merged**